From: Ben Hutchings <ben@decadent.org.uk>
Subject: audit: Increase maximum number of names logged per syscall to 30
Date: Sat, 15 Dec 2012 01:03:52 +0000
Bug-Debian: http://bugs.debian.org/631799
Forwarded: not-needed

Increase maximum number of names logged per syscall to 30, as the
current value of 20 can be too small.  This was enough for Rik Theys
who reported #631799.

There is a better but bigger fix upstream, commit 5195d8e217a7
('audit: dynamically allocate audit_names when not enough space is in
the names array').

Index: linux-3.2.46/kernel/auditsc.c
===================================================================
--- linux-3.2.46.orig/kernel/auditsc.c	2013-07-26 18:53:47.000000000 +0000
+++ linux-3.2.46/kernel/auditsc.c	2013-07-26 19:25:42.000000000 +0000
@@ -72,7 +72,7 @@
 
 /* AUDIT_NAMES is the number of slots we reserve in the audit_context
  * for saving names from getname(). */
-#define AUDIT_NAMES    20
+#define AUDIT_NAMES    30
 
 /* Indicates that audit should log the full pathname. */
 #define AUDIT_NAME_FULL -1
